src/plugins/cosmos/components/modals/Staking/Staking.tsx

```diff
@@ -58,15 +123,72 @@ const StakingModalContent = ({ assetId, action }: StakingModalProps) => {
   }
 
   const handleClose = () => {
-    history.goBack()
+    history.push(StakeRoutes.Overview)
     close()
   }
 
-  const renderStakingRoutes = () => {
-    if (action === StakingAction.Stake)
-      return (
-        <>
-          <Route exact path={StakeRoutes.Stake}>
+  const asset = (_ => ({
+    name: 'Osmosis',
+    symbol: 'OSMO',
+    caip19: assetId
+  }))(assetId) as Asset
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClose={handleClose}
+      isCentered
+      initialFocusRef={initialRef}
+      variant={isConfirmStep ? 'fluid' : ''}
+    >
+      <ModalOverlay />
+      <ModalContent width='100%' maxWidth='440px'>
+        <ModalHeader
+          textAlign='center'
+          bg={headerBg}
+          borderBottomWidth='1px'
+          borderColor={headerBorder}
+          borderTopRadius='xl'
+        >
+          <Text
+            translation={
+              !isClaimStep ? ['defi.assetStaking', { assetName: asset.symbol }] : 'defi.claimAsset'
+            }
+            pb={isClaimStep ? '12px' : '0'}
+          />
+          {!isClaimStep && (
+            <CosmosActionButtons
+              asset={asset}
+              activeRoute={
+                isStakeStep
```

Comment:
Not sure about this one. Looks like you could just pass the `stakeRoute` directly, which effectively means you can pass nothing since we can `useHistory`.
Either way, that's another nested ternary we probably don't want to have.